client/src/components/NewShoeForm.js

```diff
@@ -0,0 +1,87 @@
+import React, { useState } from "react";
+import { Redirect } from "react-router-dom";
+
+import translateServerErrors from "./../services/translateServerErrors"
+
+const NewShoeForm = (props) => {
+	const defaultForm = {
+		name: "",
+		color: "",
+		category: "",
+		url: ""
+	}
+	const [newShoe, setNewShoe] = useState(defaultForm)
+	const [errors, setErrors] = useState([])
+	const [shouldRedirect, setShouldRedirect] = useState(false)
+
+	const addNewShoe = async () => {
+		try {
+			const response = await fetch('/api/v1/shoes', {
+				method: "POST",
+				headers: new Headers({
+					"Content-Type": "application/json"
+				}),
+				body: JSON.stringify(newShoe)
+			})
+			if (!response.ok) {
+				if (response.status === 422) {
+					const body = await response.json()
+					const newErrors = translateServerErrors(body.errors)
+					return setErrors(newErrors)
+				} else {
+					const errorMessage = `${response.status} (${response.statusText})`
+					const error = new Error(errorMessage)
+					throw error
+				}
+			} else {
+				const body = await response.json()
+				console.log("Shoe created successfully.", body)
+				setShouldRedirect(true)
+			}
+		} catch (err) {
+			console.error(`Error in fetch: ${err.message}`)
+		}
+	}
```

Comment:
please work with your group to edit your project's tab spacing

e.g. https://stackoverflow.com/questions/29972396/how-can-i-customize-the-tab-to-space-conversion-factor

this indentation is too large, we need to trim this back a smidge